elyra/pipeline/processor_airflow.py

```diff
@@ -197,7 +197,14 @@ def _cc_pipeline(self, pipeline, pipeline_name):
                             image_instance.metadata.get('pull_policy'):
                         image_pull_policy = image_instance.metadata['pull_policy']
 
-                target_op = {'notebook': operation.name,
+                unique_name_counter = 1
+                unique_operation_name = operation.name
+                names = [op['notebook'] for op in target_ops]
+                while unique_operation_name in names:
+                    unique_name_counter += 1
+                    unique_operation_name = ''.join([unique_operation_name, '_', str(unique_name_counter)])
+
+                target_op = {'notebook': unique_operation_name,
```

Comment:
In your example `name` and `task_id` were updated to generate a unique name:
```
    name="load_data_2",
    namespace="default",
    task_id="load_data_2",
```

Is `operation. id` (line below) not the same as `task_id` and should therefore also be updated?

Reply:
operation.id is a UUID generated by the frontend (i think) that we use to give the notebook op instance variable a distinct name in the template. I dont think we use it anywhere else.  e.g. op_2f3f6243_82ef_43ee_af09_9888a5dfbc30

Reply:
airflow runtime specific components also appear to use a different naming scheme that uses a timestamp to generate a unique task name for the DAG.